### DISCUSSION TITLE
Shift + Up/Down/Left/Right for selection and Ctrl + Left/Right for fwd/back word

### DISCUSSION THREAD

**Discussion by jgilardi** [helix-editor/helix#7148] [Q&A] [answered]
Just getting my sea legs and trying to parse through documentation and configuration.  Maybe someone knows the answers to this and can help my onramping:

Many editors support Ctrl + Left/Right to move forward/back words.  For example, I'm using it here in Chrome to help author the text I'm writing in this input box.  Is there a way to enable this in Helix?  Out of the box Ctrl+Left/Right arrow do nothing.

Similarly supported, though less so: Shift + Up/Down/Left/Right will begin a selection/highlight (though less so in modal editors).  

In most cases you can even combine the two:  Shift + Ctrl + Left/Right will select back/forward words.

It'd be nice to be able to do this in Helix as it means it'll have parity between other applications.  If you need a comparison point to see what I'm talking about, try out the commands in VSCode.

Thanks!

**Comment by kirawi** [comment, marked as answer]
You can create a binding for this in your `config.toml`: https://docs.helix-editor.com/remapping.html

It's not possible to move without selecting, because that's part of the editing model. Instead, you can have the binding also clear the selection.

**Reply by jgilardi** [reply]
The link hiding @ the bottom of the remapping page was the key:  https://github.com/helix-editor/helix/blob/master/helix-term/src/commands.rs

The master list of commands to invoke was the piece I was missing.  I'll play around with those.  Thanks!

**Comment by danielrode** [comment]
Add this to your config (takes care of the right/left word navigation):

```
[keys.normal]
C-right = "move_next_word_start"
C-left = "move_prev_word_end"

[keys.insert]
C-right = "move_next_word_start"
C-left = "move_prev_word_end"
```

**Reply by eliliam** [reply]
Do you know how to make this work in the `SELECT` mode? Adding the below block to my config: 

```
[keys.select]
C-right = "move_next_word_start"
C-left = "move_prev_word_end"
```

This causes the visual selection to reset with each word jumped. This makes sense, but isn't the ideal behavior in the visual `SELECT` mode. Is there another action that would extend the selection forward/backward by one word we could use instead of `move_next_word_start`/`move_prev_word_end`?

Edit:
After playing around with it, I've found a working config:
```
[keys.select]
C-right = "extend_next_word_start"
C-left = "extend_prev_word_end"
```

With special thanks to @sommerper in https://github.com/helix-editor/helix/discussions/7148#discussioncomment-12212779 for the additional mappings, here is my full final version if anyone would like to use it as well:
```
[keys.normal]
C-right = "move_next_word_start"
C-left = "move_prev_word_end"
S-right = "extend_char_right"
S-left = "extend_char_left"

[keys.insert]
C-right = "move_next_word_start"
C-left = "move_prev_word_end"
S-right = "extend_char_right"
S-left = "extend_char_left"

[keys.select]
C-right = "extend_next_word_start"
C-left = "extend_prev_word_end"
S-right = "extend_char_right"
S-left = "extend_char_left"
```
I did not add the `ctrl+shift+{Left,Right}` shortcuts as those conflict with ones I use for my Kitty pane management, but if you would like to use them, refer to https://github.com/helix-editor/helix/discussions/7148#discussioncomment-12212779

**Comment by databasedav** [comment]
does anyone have a full port of this behavior to helix?

**Reply by sommerper** [reply]
```
C-right = "move_next_word_start"
C-left = "move_prev_word_end"
C-S-right = "extend_next_word_end"
C-S-left = "extend_prev_word_start"
S-right = "extend_char_right"
S-left = "extend_char_left"
```
